File: Algorithms/929. Unique Email Addresses/swift/solution.swift
class Solution {
    func numUniqueEmails(_ emails: [String]) -> Int {
        guard emails.count > 0 else { return 0 }
        var resultSet = Set<String>()

        for email in emails {
            let seperated = email.split(separator: "@")
            let local = seperated[0], validDomain = seperated[1]
            var validLocal = String()

            for char in local where char != "." {
                guard char != "+" else { break }
                validLocal.append(char)
            }

            let validFormat = validLocal + "@" + validDomain
            resultSet.insert(String(validFormat))
        }
        return resultSet.count
    }
}

File: Algorithms/929. Unique Email Addresses/javascript/solution.js
/**
 * @param {string[]} emails
 * @return {number}
 */
var numUniqueEmails = function(emails) {
    return new Set(
        emails.map(
            (mail) =>
                `${mail.split('@')[0].replace(/\+.*$|\./g, '')}@${
                    mail.split('@')[1]
                }`
        )
    ).size;
};
